functions/src/modules/pm/categories/updateCategory.ts

```diff
@@ -1,5 +1,72 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import {
+  expressErrorHandlerFactory,
+  requestBodyTypeValidator,
+} from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const updateCategory: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: validateOrganizer },
+    sponsor: { accepted: executeIfSponsorMatches },
+  })(req, res, next);
+};
 
-export default replaceMe;
+const validateOrganizer: ExpressFunction = (req, res, next) => {
+  const validationRules: ValidatorObjectRules = {
+    type: "object",
+    rules: {
+        prizeCategoryName: { rules: ["string"], required: true },
+        prizeCategoryDescription: { rules: ["string"]},
+        approvalStatus: { rules: ["string"]},
+        eligibility: { rules: ["string"]},
+        companyId: { rules: ["string"], required: true },
+    },
+  };
+  requestBodyTypeValidator(req, res, next)(validationRules, execute);
+};
+
+const validateSponsor: ExpressFunction = (req, res, next) => {
+    const validationRules: ValidatorObjectRules = {
+      type: "object",
+      rules: {
+          prizeCategoryName: { rules: ["string"], required: true },
+          prizeCategoryDescription: { rules: ["string"]},
+          approvalStatus: { rules: ["string"]},
+          eligibility: { rules: ["string"]},
+      },
+    };
+    requestBodyTypeValidator(req, res, next)(validationRules, execute);
+  };
+
+const execute: ExpressFunction = (req, res, next) => {
+  const errorHandler = expressErrorHandlerFactory(req, res, next);
+
+  firebaseApp
+    .firestore()
+    .collection("prizeCategories")
+    .doc(req.params.prizeCategoryId)
```

Comment:
Once again, we need to check what the parameter id is; I think it is just `categoryId`.
```suggestion
    .doc(req.params.categoryId)
```